### DISCUSSION TITLE
How to read blob uri?

### DISCUSSION THREAD

**Discussion by neciszhang** [donmccurdy/glTF-Transform#847] [Q&A] [answered]
Hi,
I upload a glb file, I use "URL.createObjectURL(file)" to create a uri, WebIO.readAsJSON can't read it.

**Comment by donmccurdy** [comment, marked as answer]
Is this a .glb file? If you have a [File](https://developer.mozilla.org/en-US/docs/Web/API/File) instance already, then accessing it through an ArrayBuffer and parsing that might be the way to go:

```javascript
const arrayBuffer = await file.arrayBuffer();
const { json, resources } = io.binaryToJSON(arrayBuffer);
```

If it's a .gltf and already in local memory, then you'll have to construct the [JSONDocument](https://gltf-transform.donmccurdy.com/interfaces/core.jsondocument.html) yourself in order to parse that. Relative references to textures and .bin files will not work when encoding a blob URI.

**Reply by neciszhang** [reply]
Thanks! It's work.
`const arrayBuffer = await file.arrayBuffer();`
`const uint8View = new Uint8Array(arrayBuffer);`
`const { json, resources } = await io.binaryToJSON(uint8View);`
